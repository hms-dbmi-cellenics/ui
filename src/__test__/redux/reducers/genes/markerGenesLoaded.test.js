import markerGenesLoadedReducer from 'redux/reducers/genes/markerGenesLoaded';

describe('markerGenesLoaded', () => {
  it('returns correct state', () => {
    const initialState = {
      expression: {
        views: {},
        data: {},
      },
      markers: {},
    };

    const genes = ['geneA'];
    const data = {
      geneA: {
        rawExpression: {
          expression: [1],
          mean: 1,
          stdev: 1,
        },
      },
    };

    const newState = markerGenesLoadedReducer(initialState, {
      payload: {
        genes,
        data,
        plotUuid: 'interactiveHeatmap',
      },
    });

    expect(newState).toMatchSnapshot();
  });
});
