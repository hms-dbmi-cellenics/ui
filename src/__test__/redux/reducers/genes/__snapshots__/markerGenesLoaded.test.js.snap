// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markerGenesLoaded returns correct state 1`] = `
{
  "expression": {
    "downsampled": {
      "ETag": null,
      "cellOrder": [
        0,
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
      ],
      "error": false,
      "loading": [],
      "matrix": ExpressionMatrix {
        "geneIndexes": {
          "geneA": 0,
        },
        "rawGeneExpressions": {
          "datatype": undefined,
          "index": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
          ],
          "mathjs": "SparseMatrix",
          "ptr": [
            0,
            9,
          ],
          "size": [
            10,
            1,
          ],
          "values": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
          ],
        },
        "setGeneExpression": [Function],
        "stats": {
          "rawMean": [
            1,
          ],
          "rawStdev": [
            0,
          ],
          "truncatedMax": [
            1,
          ],
          "truncatedMin": [
            1,
          ],
        },
        "truncatedGeneExpressions": {
          "datatype": undefined,
          "index": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
          ],
          "mathjs": "SparseMatrix",
          "ptr": [
            0,
            9,
          ],
          "size": [
            10,
            1,
          ],
          "values": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
          ],
        },
        "zScore": {
          "datatype": undefined,
          "index": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
          ],
          "mathjs": "SparseMatrix",
          "ptr": [
            0,
            9,
          ],
          "size": [
            10,
            1,
          ],
          "values": [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
          ],
        },
      },
    },
    "full": {
      "error": false,
      "loading": [],
      "matrix": ExpressionMatrix {
        "geneIndexes": {},
        "rawGeneExpressions": {
          "datatype": undefined,
          "index": [],
          "mathjs": "SparseMatrix",
          "ptr": [
            0,
          ],
          "size": [
            0,
            0,
          ],
          "values": [],
        },
        "setGeneExpression": [Function],
        "stats": {
          "rawMean": [],
          "rawStdev": [],
          "truncatedMax": [],
          "truncatedMin": [],
        },
        "truncatedGeneExpressions": {
          "datatype": undefined,
          "index": [],
          "mathjs": "SparseMatrix",
          "ptr": [
            0,
          ],
          "size": [
            0,
            0,
          ],
          "values": [],
        },
        "zScore": {
          "datatype": undefined,
          "index": [],
          "mathjs": "SparseMatrix",
          "ptr": [
            0,
          ],
          "size": [
            0,
            0,
          ],
          "values": [],
        },
      },
    },
    "views": {
      "interactiveHeatmap": {
        "data": [
          "geneA",
        ],
        "error": false,
        "fetching": false,
        "markers": true,
      },
    },
  },
  "focused": undefined,
  "markers": {
    "ETag": null,
    "error": false,
    "loading": false,
  },
  "properties": {
    "data": {},
    "loading": [],
    "views": {},
  },
  "selected": [],
}
`;
